app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,80 @@
+class CategoriesController < ApplicationController
+  before_action :load_category, only: [:show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token
+  # GET categories/new
+  def new
+  	@category = Category.new
+
+    respond_to do |format|
+      format.html
+    end
+  end
+
+  # POST categories/
+  def create
+  	@category = Category.create(category_params)
+    if @category.valid?
+      flash[:notice] = 'Category has been created!'
+      redirect_to category_path(@category)
+    else
+      flash[:error] = @category.errors.full_messages
+      redirect_to categories_path
```

Comment:
no need to redirect to listing_path, re render new form on failure

Reply:
removed